### PR TITLE
SF-3750: Explicit Data Includes

### PR DESCRIPTION
**Pull request by quellen-sol**
- [x] SQUASH

### CODE REVIEW

File: core/src/banking_stage/committer.rs [hunk 1]
```diff
@@ -8,8 +8,8 @@ use {
     solana_runtime::{
         accounts::TransactionLoadResult,
         bank::{
-            Bank, CommitTransactionCounts, TransactionBalancesSet, TransactionExecutionResult,
-            TransactionResults, TransactionDatumSet
+            Bank, CommitTransactionCounts, PreOrPostDatum, TransactionBalancesSet, TransactionDatumSet,
```
Comment (thesoftwarejedi) [inline core/src/banking_stage/committer.rs]: for mergeability, put additions in a line by themselves
realizing that I didn't even do this.  so don't worry about it unless you want ot figure out how this line looked before, put it that way, and add our additions below.  meh

File: core/src/banking_stage/consumer.rs [hunk 2]
```diff
@@ -20,7 +20,7 @@ use {
     },
     solana_program_runtime::timings::ExecuteTimings,
     solana_runtime::{
-        bank::{Bank, LoadAndExecuteTransactionsOutput, TransactionCheckResult},
+        bank::{Bank, LoadAndExecuteTransactionsOutput, PreOrPostDatum, TransactionCheckResult},
```
Comment (thesoftwarejedi) [inline core/src/banking_stage/consumer.rs]: for mergeability, put additions on own line

File: ledger/src/blockstore_processor.rs [hunk 3]
```diff
@@ -28,8 +28,8 @@ use {
         accounts_index::AccountSecondaryIndexes,
         accounts_update_notifier_interface::AccountsUpdateNotifier,
         bank::{
-            Bank, TransactionBalancesSet, TransactionDatumSet, TransactionExecutionDetails, TransactionExecutionResult,
-            TransactionResults,
+            Bank, TransactionBalancesSet, TransactionDatumSet, TransactionExecutionDetails,
```
Comment (thesoftwarejedi) [inline ledger/src/blockstore_processor.rs:31]: for mergeability, put additions on own line

File: runtime/src/bank.rs [hunk 4]
```diff
@@ -1091,9 +1096,12 @@ pub struct Bank {
 
     epoch_reward_status: EpochRewardStatus,
     /// programs that will not have their account data sent to geyser
-    pub dataum_excluded_programs: (HashSet<Pubkey>, HashSet<Pubkey>),
+    pub program_datum_inclusions: ProgramDatumInclusions,
 }
 
+/// HashMap<Pubkey, (include_pre: bool, include_post: bool)>;
+pub type ProgramDatumInclusions = HashMap<Pubkey, (bool, bool)>;
```
Comment (thesoftwarejedi) [inline runtime/src/bank.rs]: make a struct for the tuple; cuz why not

File: runtime/src/bank.rs [hunk 5]
```diff
@@ -6058,20 +6043,22 @@ impl Bank {
...
-            || self.dataum_excluded_programs.1.contains(key) {
+        let inclusion = self.program_datum_inclusions.get(owner)?;
+        let include_data = match pre_or_post {
+            PreOrPostDatum::PreDatum => inclusion.0,
+            PreOrPostDatum::PostDatum => inclusion.1
+        };
+
+        // no data for
+        // large accounts
+        if !include_data || data.len() > STEP_TX_DATUM_MAX_SIZE || account.executable() {
```
Comment (thesoftwarejedi) [inline runtime/src/bank.rs]: lets kill this off too

File: runtime/src/runtime_config.rs [hunk 6]
```diff
@@ -1,9 +1,12 @@
 use solana_program_runtime::compute_budget::ComputeBudget;
 
+use crate::bank::ProgramDatumInclusions;
+
 /// Encapsulates flags that can be used to tweak the runtime behavior.
```
Comment (thesoftwarejedi) [inline runtime/src/runtime_config.rs:7]: this isn't the right file.  theres a geyser config file that it reads somewhere somehow

**Review by thesoftwarejedi — CHANGES_REQUESTED** [review]
killin it

**Review by JamesMDunn — APPROVED** [review]
🦀 🦀 🦀 🦀 🦀

**Review by thesoftwarejedi — APPROVED** [review]
Where is this data compressed?  Doesn't it zip it somewhere?  Is that in our plugin perhaps?
Looks good.

**Comment by thesoftwarejedi** [comment]
I assume tested locally with our plugin with success.

**Comment by quellen-sol** [comment]
> I assume tested locally with our plugin with success.

Yup tested with a local ingestooor and everything :+1:

**Comment by quellen-sol** [comment]
> Where is this data compressed? Doesn't it zip it somewhere? Is that in our plugin perhaps? Looks good.

Plugin does it I think

**Pull request merged**